File: android/feature/auth/src/main/java/com/enterprise/feature/auth/domain/entities/AuthState.kt
package com.enterprise.feature.auth.domain.entities

import com.enterprise.core.domain.entities.DomainError
import com.enterprise.core.domain.entities.User

/** Authentication state for the Auth feature MVI pattern. */
sealed class AuthState {
    data object Idle : AuthState()
    data object Authenticating : AuthState()
    data class Authenticated(val user: User) : AuthState()
    data class Failed(val error: DomainError) : AuthState()
}

File: ios/EnterpriseMobileApp/Sources/Feature/Auth/Domain/Entities/AuthState.swift
import Foundation
import CoreDomain

/// Authentication state for the Auth feature module.
public enum AuthState: Equatable, Sendable {
    case idle
    case authenticating
    case authenticated(User)
    case failed(DomainError)
}

/// Credentials for email/password login.
public struct LoginCredentials: Equatable, Sendable {
    public let email: String
    public let password: String

    public init(email: String, password: String) {
        self.email = email
        self.password = password
    }
}

/// Biometric authentication type.
public enum BiometricType: Equatable, Sendable {
    case faceID
    case touchID
    case none
}
